src/routes/dashboard/admin/time-slots/+page.svelte

```
@@ -0,0 +1,232 @@
+    import NavigationDropDown from 'elements/navigation/NavigationDropDown.svelte';
+    import UnsavedChangesCallbackWrapper from 'elements/navigation/unsavedChangesCallbackWrapper.svelte';
+    import Button from 'elements/input/button.svelte';
+    import Input from 'elements/input/input.svelte';
+    import Icon from 'elements/image/icon.svelte';
+    import Toggle from 'elements/input/toggle.svelte';
+    import SaveMessage from 'elements/text/saveMessage.svelte';
+    import Message from 'elements/text/message.svelte';
+
+    export let data: LoadAdminTimeSlots;
+
+    let saveMessage: SaveMessage;
+    let errorMessage: string | undefined = undefined;
+
+    function getIDFromTitle(title: string): number {
+        for (const entry of data.allEvents) {
+            if (entry.title === title) {
+                return entry.id;
+            }
+        }
+
+        console.log(`not able to look up ID for ${title}`);
+        throw error(404);
+    }
+
+    function getEventFromID(id: number): DashboardEvent {
+        for (const event of data.allEvents) {
+            if (event.id === id) {
+                return event;
+            }
+        }
+
+        console.log(`not able to look up Event with ID ${id}`);
+        throw error(404);
+    }
+
+    function addSlot() {
+        const newSlotsData: DashboardTimeSlot = {
+            event_id:   data.currentEventID,
+            id:         0,
+            start_time: getEventFromID(data.currentEventID).start_date + 'T00:00:00', // the start of the event is used here, as it should not be too far away from the actual date.
+            duration:   0,
+            is_special: false,
+        };
+
+        data.currentSlots = [
+            ...data.currentSlots,
+            newSlotsData,
+        ];
+    }
+
+    async function updateDisplayed(value: string) {
+        data.currentEventID = getIDFromTitle(value);
+        data.currentSlots   = await getTimeSlotsAsync(fetch, data.currentEventID);
+    }
```

I think you should also hide the error message here. Otherwise it will stay visible even after changing to a different event.